pkg/stack/loader.go

```diff
@@ -36,50 +34,44 @@ func LoadStacks(dir string, stackService Service) error {
 	if err != nil {
 		return fmt.Errorf("error in stack config: %v", err)
 	}
-	_ = stacks
-
-	entries, err := os.ReadDir(dir)
-	if err != nil {
-		return fmt.Errorf("failed to read stack folder: %s", err)
-	}
-
-	for _, entry := range entries {
-		if !entry.IsDir() {
-			continue
-		}
 
-		name := entry.Name()
-		log.Printf("Parsing stack: %q\n", name)
-		stackTemplate, err := parseStack(dir, name)
-		if err != nil {
-			return fmt.Errorf("error parsing stack %q: %v", name, err)
-		}
+	for name := range stacks {
+		// TODO: If I'm not doing the below I'm getting: G601 (CWE-118): Implicit memory aliasing in for loop. (Confidence: MEDIUM, Severity: MEDIUM)
+		// An alternative to the below would be to ignore the warning, but I'm not sure what's best
+		stack := stacks[name]
 
-		existingStack, err := stackService.Find(name)
+		_, err := stackService.Find(name)
 		if err != nil {
 			if !errdef.IsNotFound(err) {
 				return fmt.Errorf("error searching existing stack %q: %w", name, err)
 			}
 		}
 		if err == nil {
-			log.Printf("Stack exists: %s\n", existingStack.Name)
+			log.Printf("Stack exists: %s\n", name)
 			// TODO: For now just bail if the stack exists. This should probably be done differently so we can reload the stack if it has changed
 			// If we have running instances we can't just change parameters etc. though
 			continue
 		}
 
-		stack := &model.Stack{
-			Name:             name,
-			HostnamePattern:  stackTemplate.hostnamePattern,
-			HostnameVariable: stackTemplate.hostnameVariable,
-			Parameters:       make(map[string]model.StackParameter),
+		parsedStack, err := parseStack(dir, name)
+		if err != nil {
+			return fmt.Errorf("error parsing stack %q: %w", name, err)
 		}
-		for name, v := range stackTemplate.parameters {
-			isConsumed := isConsumedParameter(name, stackTemplate.consumedParameters)
-			stack.Parameters[name] = model.StackParameter{Consumed: isConsumed, DefaultValue: v}
+
+		parsedParameters := len(maps.Keys(parsedStack.parameters))
+		stackParameters := len(stack.Parameters)
+		if parsedParameters != stackParameters {
+			return fmt.Errorf("number of stack parameters (%d) doesn't match helmfile parameters (%d) for stack: %q", stackParameters, parsedParameters, name)
```

Comment:
loader_test.go creates a diff between helmfile parameters and the ones in the stack.go file so there is no need to do this here 😄 

We could move the parsing of stacks into loader_test.go entirely.